contracts/examples/IncentivizedFlatDirectory.sol

```diff
@@ -0,0 +1,74 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "./FlatDirectory.sol";
+
+contract IncentivizedFlatDirectory is FlatDirectory {
+    address public operator;
+
+    uint256 public constant PER_CHUNK_SIZE = 24 * 1024;
+    uint256 public constant CODE_STAKING_PER_CHUNK = 10**18;
+
+    constructor(uint8 _slotLimit) payable FlatDirectory(_slotLimit) {}
+
+    modifier onlyOperatorOrOwner() {
+        require(operator == msg.sender || owner == msg.sender, "must from owner or operator");
+        _;
+    }
+
+    receive() external payable {
+        if (msg.value > 0) {
+            require(msg.sender == owner, "only owner transfer token into contract");
+        }
+    }
+
+    function changeOperator(address _operator) public virtual onlyOperatorOrOwner {
+        operator = _operator;
+    }
+
+    function setDefault(bytes memory _defaultFile) public virtual override onlyOperatorOrOwner {
+        defaultFile = _defaultFile;
+    }
+
+    function write(bytes memory name, bytes memory data) public payable override onlyOperatorOrOwner {
+        require(msg.value == 0, "msg.value must be 0");
```

Comment:
I would suggest using calldata in https://github.com/QuarkChain/evm-large-storage/pull/6 to save gas cost.

You can rebase the code on 6 or create another issue to address it.